presto-spi/src/main/java/com/facebook/presto/spi/type/Chars.java

```diff
@@ -95,6 +105,36 @@ private static int compareCharsShorterToLonger(Slice shorter, Slice longer)
         return 0;
     }
 
+    private static int compareCharsShorterToLongerNoPad(Slice shorter, Slice longer, int shorterLength, int longerLength)
+    {
+        for (int i = 0; i < shorterLength; ++i) {
+            int result;
+            if (i >= shorter.length()) {
+                if (i >= longer.length()) {
+                    return 0;
```

Comment:
this is not correct to same prefix slices, but different type length. I've fixed that